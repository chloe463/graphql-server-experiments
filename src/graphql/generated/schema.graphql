### This file was generated by Nexus Schema
### Do not make changes to this file directly


type CancelToDeleteQuestionnairePayload {
  questionnaire: Questionnaire
}

type Comment {
  body: String!
  email: String!
  id: Int!
  name: String!
  postId: Int!
}

input CreateOptionInput {
  text: String!
}

input CreateQuestionInput {
  options: [CreateOptionInput]
  required: Boolean!
  text: String!
  type: Int!
}

input CreateQuestionnaireInput {
  description: String!
  endAt: DateTime!
  questions: [CreateQuestionInput]
  startAt: DateTime!
  state: Int
  title: String!
}

type CreateQuestionnairePayload {
  questionnaire: Questionnaire
}

input CreateTodoInput {
  task: String!
}

type CreateTodoPayload {
  todo: Todo
}

scalar Date

scalar DateTime

input DeleteQuestionnaireInput {
  id: Int!
}

type DeleteQuestionnairePayload {
  id: Int
  result: Boolean
}

type Mutation {
  cancelToDeleteQuestionnaire(id: Int!): CancelToDeleteQuestionnairePayload
  createQuestionnaire(questionnaire: CreateQuestionnaireInput): CreateQuestionnairePayload
  createTodo(todo: CreateTodoInput): CreateTodoPayload
  deleteQuestionnaire(id: Int!): DeleteQuestionnairePayload
  updateQuestionnaire(questionnaire: UpdateQuestionnaireInput!): UpdateQuestionnairePayload
  updateTodo(todo: UpdateTodoInput): UpdateTodoPayload
}

type Option {
  id: Int!
  text: String!
}

"""
PageInfo cursor, as defined in https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo
"""
type PageInfo {
  """
  The cursor corresponding to the last nodes in edges. Null if the connection is empty.
  """
  endCursor: String

  """
  Used to indicate whether more edges exist following the set defined by the clients arguments.
  """
  hasNextPage: Boolean!

  """
  Used to indicate whether more edges exist prior to the set defined by the clients arguments.
  """
  hasPreviousPage: Boolean!

  """
  The cursor corresponding to the first nodes in edges. Null if the connection is empty.
  """
  startCursor: String
}

type Post {
  body: String!
  id: Int!
  title: String!
  userId: Int!
}

type PostEdge {
  """https://facebook.github.io/relay/graphql/connections.htm#sec-Cursor"""
  cursor: String!

  """https://facebook.github.io/relay/graphql/connections.htm#sec-Node"""
  node: Post!
}

type Query {
  comments(postId: Int!): [Comment]!
  postConnection(
    """Returns the elements in the list that come after the specified cursor"""
    after: String

    """Returns the elements in the list that come before the specified cursor"""
    before: String

    """Returns the first n elements from the list."""
    first: Int

    """Returns the last n elements from the list."""
    last: Int

    """Dummy arg"""
    query: String
  ): QueryPostConnection_Connection!
  posts(limit: Int = 10, start: String = "0"): [Post]
  questionnaire(id: Int): Questionnaire
  questionnaireConnection(
    """Returns the elements in the list that come after the specified cursor"""
    after: String

    """Returns the elements in the list that come before the specified cursor"""
    before: String

    """Returns the first n elements from the list."""
    first: Int

    """Returns the last n elements from the list."""
    last: Int
  ): QueryQuestionnaireConnection_Connection!
  questionnaires: [Questionnaire]
  todoConnection(
    """Returns the elements in the list that come after the specified cursor"""
    after: String

    """Returns the elements in the list that come before the specified cursor"""
    before: String

    """Returns the first n elements from the list."""
    first: Int

    """Returns the last n elements from the list."""
    last: Int
  ): QueryTodoConnection_Connection!
}

type QueryPostConnection_Connection {
  """
  https://facebook.github.io/relay/graphql/connections.htm#sec-Edge-Types
  """
  edges: [PostEdge!]!

  """
  https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo
  """
  pageInfo: PageInfo!
  totalCount: Int!
}

type QueryQuestionnaireConnection_Connection {
  """
  https://facebook.github.io/relay/graphql/connections.htm#sec-Edge-Types
  """
  edges: [QuestionnaireEdge!]!

  """
  https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo
  """
  pageInfo: PageInfo!
  totalCount: Int!
}

type QueryTodoConnection_Connection {
  """
  https://facebook.github.io/relay/graphql/connections.htm#sec-Edge-Types
  """
  edges: [TodoEdge!]!

  """
  https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo
  """
  pageInfo: PageInfo!
  totalCount: Int!
}

type Question {
  id: Int!
  options: [Option]!
  text: String!
  type: Int!
}

type Questionnaire {
  description: String!
  endAt: DateTime!
  id: Int!
  questions: [Question]!
  startAt: DateTime!
  state: Int!
  title: String!
}

type QuestionnaireEdge {
  """https://facebook.github.io/relay/graphql/connections.htm#sec-Cursor"""
  cursor: String!

  """https://facebook.github.io/relay/graphql/connections.htm#sec-Node"""
  node: Questionnaire!
}

type Todo {
  createdAt: DateTime!
  finishedAt: DateTime
  id: Int!
  task: String!
  updatedAt: DateTime!
}

type TodoEdge {
  """https://facebook.github.io/relay/graphql/connections.htm#sec-Cursor"""
  cursor: String!

  """https://facebook.github.io/relay/graphql/connections.htm#sec-Node"""
  node: Todo!
}

input UpdateOptionInput {
  id: Int
  text: String
}

input UpdateQuestionInput {
  id: Int
  options: [UpdateOptionInput]
  required: Boolean
  text: String
  type: Int
}

input UpdateQuestionnaireInput {
  description: String
  endAt: DateTime
  id: Int!
  questions: [UpdateQuestionInput]
  startAt: DateTime
  state: Int
  title: String
}

type UpdateQuestionnairePayload {
  questionnaire: Questionnaire
}

input UpdateTodoInput {
  finishedAt: DateTime
  id: Int!
  task: String!
}

type UpdateTodoPayload {
  todo: Todo
}
