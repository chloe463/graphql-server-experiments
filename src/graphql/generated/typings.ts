/**
 * This file was generated by Nexus Schema
 * Do not make changes to this file directly
 */


import { Context } from "./../../context"
import { core, connectionPluginCore } from "nexus"
declare global {
  interface NexusGenCustomInputMethods<TypeName extends string> {
    datetime<FieldName extends string>(fieldName: FieldName, opts?: core.CommonInputFieldConfig<TypeName, FieldName>): void // "DateTime";
    date<FieldName extends string>(fieldName: FieldName, opts?: core.CommonInputFieldConfig<TypeName, FieldName>): void // "Date";
  }
}
declare global {
  interface NexusGenCustomOutputMethods<TypeName extends string> {
    datetime<FieldName extends string>(fieldName: FieldName, ...opts: core.ScalarOutSpread<TypeName, FieldName>): void // "DateTime";
    date<FieldName extends string>(fieldName: FieldName, ...opts: core.ScalarOutSpread<TypeName, FieldName>): void // "Date";
    /**
     * Adds a Relay-style connection to the type, with numerous options for configuration
     *
     * @see https://nexusjs.org/docs/plugins/connection
     */
    connectionField<FieldName extends string>(
      fieldName: FieldName,
      config: connectionPluginCore.ConnectionFieldConfig<TypeName, FieldName>
    ): void
  }
}


declare global {
  interface NexusGen extends NexusGenTypes {}
}

export interface NexusGenInputs {
  CreateOptionInput: { // input type
    text: string; // String!
  }
  CreateQuestionInput: { // input type
    options: Array<NexusGenInputs['CreateOptionInput'] | null>; // [CreateOptionInput]!
    required: boolean; // Boolean!
    text: string; // String!
    type: number; // Int!
  }
  CreateQuestionnaireInput: { // input type
    description: string; // String!
    endAt: NexusGenScalars['DateTime']; // DateTime!
    questions: Array<NexusGenInputs['CreateQuestionInput'] | null>; // [CreateQuestionInput]!
    startAt: NexusGenScalars['DateTime']; // DateTime!
    state: number; // Int!
    title: string; // String!
  }
  DeleteQuestionnaireInput: { // input type
    id: number; // Int!
  }
  UpdateOptionInput: { // input type
    id?: number | null; // Int
    text?: string | null; // String
  }
  UpdateQuestionInput: { // input type
    id?: number | null; // Int
    options?: Array<NexusGenInputs['UpdateOptionInput'] | null> | null; // [UpdateOptionInput]
    required?: boolean | null; // Boolean
    text?: string | null; // String
    type?: number | null; // Int
  }
  UpdateQuestionnaireInput: { // input type
    description?: string | null; // String
    endAt?: NexusGenScalars['DateTime'] | null; // DateTime
    id: number; // Int!
    questions?: Array<NexusGenInputs['UpdateQuestionInput'] | null> | null; // [UpdateQuestionInput]
    startAt?: NexusGenScalars['DateTime'] | null; // DateTime
    state?: number | null; // Int
    title?: string | null; // String
  }
}

export interface NexusGenEnums {
}

export interface NexusGenScalars {
  String: string
  Int: number
  Float: number
  Boolean: boolean
  ID: string
  Date: any
  DateTime: any
}

export interface NexusGenObjects {
  Comment: { // root type
    body: string; // String!
    email: string; // String!
    id: number; // Int!
    name: string; // String!
    postId: number; // Int!
  }
  CreateQuestionnairePayload: { // root type
    questionnaire?: NexusGenRootTypes['Questionnaire'] | null; // Questionnaire
  }
  DeleteQuestionnairePayload: { // root type
    result?: boolean | null; // Boolean
  }
  Mutation: {};
  Option: { // root type
    id: number; // Int!
    text: string; // String!
  }
  PageInfo: { // root type
    endCursor?: string | null; // String
    hasNextPage: boolean; // Boolean!
    hasPreviousPage: boolean; // Boolean!
    startCursor?: string | null; // String
  }
  Post: { // root type
    body: string; // String!
    id: number; // Int!
    title: string; // String!
    userId: number; // Int!
  }
  PostEdge: { // root type
    cursor: string; // String!
    node: NexusGenRootTypes['Post']; // Post!
  }
  Query: {};
  QueryPostConnection_Connection: { // root type
    edges: NexusGenRootTypes['PostEdge'][]; // [PostEdge!]!
    pageInfo: NexusGenRootTypes['PageInfo']; // PageInfo!
  }
  QueryQuestionnaireConnection_Connection: { // root type
    edges: NexusGenRootTypes['QuestionnaireEdge'][]; // [QuestionnaireEdge!]!
    pageInfo: NexusGenRootTypes['PageInfo']; // PageInfo!
  }
  Question: { // root type
    id: number; // Int!
    text: string; // String!
    type: number; // Int!
  }
  Questionnaire: { // root type
    description: string; // String!
    endAt: NexusGenScalars['DateTime']; // DateTime!
    id: number; // Int!
    questions: Array<NexusGenRootTypes['Question'] | null>; // [Question]!
    startAt: NexusGenScalars['DateTime']; // DateTime!
    state: number; // Int!
    title: string; // String!
  }
  QuestionnaireEdge: { // root type
    cursor: string; // String!
    node: NexusGenRootTypes['Questionnaire']; // Questionnaire!
  }
  UpdateQuestionnairePayload: { // root type
    questionnaire?: NexusGenRootTypes['Questionnaire'] | null; // Questionnaire
  }
}

export interface NexusGenInterfaces {
}

export interface NexusGenUnions {
}

export type NexusGenRootTypes = NexusGenObjects

export type NexusGenAllTypes = NexusGenRootTypes & NexusGenScalars

export interface NexusGenFieldTypes {
  Comment: { // field return type
    body: string; // String!
    email: string; // String!
    id: number; // Int!
    name: string; // String!
    postId: number; // Int!
  }
  CreateQuestionnairePayload: { // field return type
    questionnaire: NexusGenRootTypes['Questionnaire'] | null; // Questionnaire
  }
  DeleteQuestionnairePayload: { // field return type
    result: boolean | null; // Boolean
  }
  Mutation: { // field return type
    UpdateQuestionnaire: NexusGenRootTypes['UpdateQuestionnairePayload'] | null; // UpdateQuestionnairePayload
    createQuestionnaire: NexusGenRootTypes['CreateQuestionnairePayload'] | null; // CreateQuestionnairePayload
    deleteQuestionnaire: NexusGenRootTypes['DeleteQuestionnairePayload'] | null; // DeleteQuestionnairePayload
  }
  Option: { // field return type
    id: number; // Int!
    text: string; // String!
  }
  PageInfo: { // field return type
    endCursor: string | null; // String
    hasNextPage: boolean; // Boolean!
    hasPreviousPage: boolean; // Boolean!
    startCursor: string | null; // String
  }
  Post: { // field return type
    body: string; // String!
    id: number; // Int!
    title: string; // String!
    userId: number; // Int!
  }
  PostEdge: { // field return type
    cursor: string; // String!
    node: NexusGenRootTypes['Post']; // Post!
  }
  Query: { // field return type
    comments: Array<NexusGenRootTypes['Comment'] | null>; // [Comment]!
    postConnection: NexusGenRootTypes['QueryPostConnection_Connection']; // QueryPostConnection_Connection!
    posts: Array<NexusGenRootTypes['Post'] | null> | null; // [Post]
    questionnaireConnection: NexusGenRootTypes['QueryQuestionnaireConnection_Connection']; // QueryQuestionnaireConnection_Connection!
    questionnaires: Array<NexusGenRootTypes['Questionnaire'] | null> | null; // [Questionnaire]
  }
  QueryPostConnection_Connection: { // field return type
    edges: NexusGenRootTypes['PostEdge'][]; // [PostEdge!]!
    pageInfo: NexusGenRootTypes['PageInfo']; // PageInfo!
    totalCount: number; // Int!
  }
  QueryQuestionnaireConnection_Connection: { // field return type
    edges: NexusGenRootTypes['QuestionnaireEdge'][]; // [QuestionnaireEdge!]!
    pageInfo: NexusGenRootTypes['PageInfo']; // PageInfo!
    totalCount: number; // Int!
  }
  Question: { // field return type
    id: number; // Int!
    options: Array<NexusGenRootTypes['Option'] | null>; // [Option]!
    text: string; // String!
    type: number; // Int!
  }
  Questionnaire: { // field return type
    description: string; // String!
    endAt: NexusGenScalars['DateTime']; // DateTime!
    id: number; // Int!
    questions: Array<NexusGenRootTypes['Question'] | null>; // [Question]!
    startAt: NexusGenScalars['DateTime']; // DateTime!
    state: number; // Int!
    title: string; // String!
  }
  QuestionnaireEdge: { // field return type
    cursor: string; // String!
    node: NexusGenRootTypes['Questionnaire']; // Questionnaire!
  }
  UpdateQuestionnairePayload: { // field return type
    questionnaire: NexusGenRootTypes['Questionnaire'] | null; // Questionnaire
  }
}

export interface NexusGenFieldTypeNames {
  Comment: { // field return type name
    body: 'String'
    email: 'String'
    id: 'Int'
    name: 'String'
    postId: 'Int'
  }
  CreateQuestionnairePayload: { // field return type name
    questionnaire: 'Questionnaire'
  }
  DeleteQuestionnairePayload: { // field return type name
    result: 'Boolean'
  }
  Mutation: { // field return type name
    UpdateQuestionnaire: 'UpdateQuestionnairePayload'
    createQuestionnaire: 'CreateQuestionnairePayload'
    deleteQuestionnaire: 'DeleteQuestionnairePayload'
  }
  Option: { // field return type name
    id: 'Int'
    text: 'String'
  }
  PageInfo: { // field return type name
    endCursor: 'String'
    hasNextPage: 'Boolean'
    hasPreviousPage: 'Boolean'
    startCursor: 'String'
  }
  Post: { // field return type name
    body: 'String'
    id: 'Int'
    title: 'String'
    userId: 'Int'
  }
  PostEdge: { // field return type name
    cursor: 'String'
    node: 'Post'
  }
  Query: { // field return type name
    comments: 'Comment'
    postConnection: 'QueryPostConnection_Connection'
    posts: 'Post'
    questionnaireConnection: 'QueryQuestionnaireConnection_Connection'
    questionnaires: 'Questionnaire'
  }
  QueryPostConnection_Connection: { // field return type name
    edges: 'PostEdge'
    pageInfo: 'PageInfo'
    totalCount: 'Int'
  }
  QueryQuestionnaireConnection_Connection: { // field return type name
    edges: 'QuestionnaireEdge'
    pageInfo: 'PageInfo'
    totalCount: 'Int'
  }
  Question: { // field return type name
    id: 'Int'
    options: 'Option'
    text: 'String'
    type: 'Int'
  }
  Questionnaire: { // field return type name
    description: 'String'
    endAt: 'DateTime'
    id: 'Int'
    questions: 'Question'
    startAt: 'DateTime'
    state: 'Int'
    title: 'String'
  }
  QuestionnaireEdge: { // field return type name
    cursor: 'String'
    node: 'Questionnaire'
  }
  UpdateQuestionnairePayload: { // field return type name
    questionnaire: 'Questionnaire'
  }
}

export interface NexusGenArgTypes {
  Mutation: {
    UpdateQuestionnaire: { // args
      questionnaire: NexusGenInputs['UpdateQuestionnaireInput']; // UpdateQuestionnaireInput!
    }
    createQuestionnaire: { // args
      questionnaire?: NexusGenInputs['CreateQuestionnaireInput'] | null; // CreateQuestionnaireInput
    }
    deleteQuestionnaire: { // args
      id: number; // Int!
    }
  }
  Query: {
    comments: { // args
      postId: number; // Int!
    }
    postConnection: { // args
      after?: string | null; // String
      before?: string | null; // String
      first?: number | null; // Int
      last?: number | null; // Int
      query?: string | null; // String
    }
    posts: { // args
      limit: number | null; // Int
      start: string | null; // String
    }
    questionnaireConnection: { // args
      after?: string | null; // String
      before?: string | null; // String
      first?: number | null; // Int
      last?: number | null; // Int
    }
  }
}

export interface NexusGenAbstractTypeMembers {
}

export interface NexusGenTypeInterfaces {
}

export type NexusGenObjectNames = keyof NexusGenObjects;

export type NexusGenInputNames = keyof NexusGenInputs;

export type NexusGenEnumNames = never;

export type NexusGenInterfaceNames = never;

export type NexusGenScalarNames = keyof NexusGenScalars;

export type NexusGenUnionNames = never;

export type NexusGenObjectsUsingAbstractStrategyIsTypeOf = never;

export type NexusGenAbstractsUsingStrategyResolveType = never;

export type NexusGenFeaturesConfig = {
  abstractTypeStrategies: {
    isTypeOf: false
    resolveType: true
    __typename: false
  }
}

export interface NexusGenTypes {
  context: Context;
  inputTypes: NexusGenInputs;
  rootTypes: NexusGenRootTypes;
  inputTypeShapes: NexusGenInputs & NexusGenEnums & NexusGenScalars;
  argTypes: NexusGenArgTypes;
  fieldTypes: NexusGenFieldTypes;
  fieldTypeNames: NexusGenFieldTypeNames;
  allTypes: NexusGenAllTypes;
  typeInterfaces: NexusGenTypeInterfaces;
  objectNames: NexusGenObjectNames;
  inputNames: NexusGenInputNames;
  enumNames: NexusGenEnumNames;
  interfaceNames: NexusGenInterfaceNames;
  scalarNames: NexusGenScalarNames;
  unionNames: NexusGenUnionNames;
  allInputTypes: NexusGenTypes['inputNames'] | NexusGenTypes['enumNames'] | NexusGenTypes['scalarNames'];
  allOutputTypes: NexusGenTypes['objectNames'] | NexusGenTypes['enumNames'] | NexusGenTypes['unionNames'] | NexusGenTypes['interfaceNames'] | NexusGenTypes['scalarNames'];
  allNamedTypes: NexusGenTypes['allInputTypes'] | NexusGenTypes['allOutputTypes']
  abstractTypes: NexusGenTypes['interfaceNames'] | NexusGenTypes['unionNames'];
  abstractTypeMembers: NexusGenAbstractTypeMembers;
  objectsUsingAbstractStrategyIsTypeOf: NexusGenObjectsUsingAbstractStrategyIsTypeOf;
  abstractsUsingStrategyResolveType: NexusGenAbstractsUsingStrategyResolveType;
  features: NexusGenFeaturesConfig;
}


declare global {
  interface NexusGenPluginTypeConfig<TypeName extends string> {
  }
  interface NexusGenPluginFieldConfig<TypeName extends string, FieldName extends string> {
    
  }
  interface NexusGenPluginInputFieldConfig<TypeName extends string, FieldName extends string> {
  }
  interface NexusGenPluginSchemaConfig {
  }
  interface NexusGenPluginArgConfig {
  }
}