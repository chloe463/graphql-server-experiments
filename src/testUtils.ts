import { ApolloServer } from "apollo-server";
import { Context } from "./context";
import { schema } from "./graphql/schema";

type DummyContext = {
  [key in keyof Context]: jest.Mock;
};

export const constructTestServer = (context: DummyContext) => {
  const server = new ApolloServer({
    schema,
    context,
  });

  return server;
};
