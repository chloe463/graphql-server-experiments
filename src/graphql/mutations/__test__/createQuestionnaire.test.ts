import { PrismaClient } from "@prisma/client";
import { gql } from "apollo-server";
import { createTestClient } from "apollo-server-testing";
import DataLoader from "dataloader";
import { constructTestServer } from "../../../testUtils";
import { NexusGenInputs, NexusGenObjects } from "../../generated/typings";

const prismaClientMock = (PrismaClient as any) as jest.Mock<PrismaClient>;
const optionsLoaderMock = (DataLoader as any) as jest.Mock<DataLoader<number, any, number>>

type Variables = NexusGenInputs["CreateQuestionnaireInput"];
type Result = NexusGenObjects["CreateQuestionnairePayload"]["questionnaire"];

const CREATE_QUESTIONNAIRE_MUTATION = gql`
  mutation CreateQuestionnaire($questionnaire: CreateQuestionnaireInput!) {
    createQuestionnaire(questionnaire: $questionnaire) {
      questionnaire {
        id
        title
        description
        state
        startAt
        endAt
        questions {
          id
          text
          type
          options {
            id
            text
          }
        }
      }
    }
  }
`;

describe("[Mutation] createQuestionnaire", () => {
  it("can create new questionnaire", async () => {
    const variables: Variables = {
      title: "Test questionnaire",
      description: "inventore eum sed laboriosam qui repudiandae in eum quidem illo",
      state: 1,
      startAt: "2021-01-01 00:00:00",
      endAt: "2021-01-31 23:59:59",
      questions: [
        {
          type: 1,
          text: "Question1",
          required: false,
          options: [
            { text: "Option1-1" },
            { text: "Option1-2" },
          ],
        },
        {
          type: 2,
          text: "Question2",
          required: true,
          options: [
            { text: "Option2-1" },
            { text: "Option2-2" },
          ],
        },
      ],
    };
    (prismaClientMock as any).questionnaire = {
      create: jest.fn().mockReturnValue({
        id: 1,
        title: "Test questionnaire",
        description: "inventore eum sed laboriosam qui repudiandae in eum quidem illo",
        state: 1,
        startAt: new Date("2021-01-01 00:00:00"),
        endAt: new Date("2021-01-31 23:59:59"),
        questions: [
          {
            id: 1,
            type: 1,
            text: "Question1",
            required: false,
            options: [
              { id: 1, text: "Option1-1" },
              { id: 2, text: "Option1-2" },
            ],
          },
          {
            id: 2,
            type: 2,
            text: "Question2",
            required: true,
            options: [
              { id: 1, text: "Option2-1" },
              { id: 2, text: "Option2-2" },
            ],
          },
        ],
      }),
    };
    (optionsLoaderMock as any).load = (id: number) => {
      return [
        { id: 1, text: `Option${id}-1` },
        { id: 2, text: `Option${id}-2` },
      ];
    };
    const server = constructTestServer({
      prismaClient: prismaClientMock,
      optionsLoader: optionsLoaderMock,
    });

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: CREATE_QUESTIONNAIRE_MUTATION,
      variables: {
        questionnaire: variables,
      },
    });
    expect(res).toMatchSnapshot();
  });
});
