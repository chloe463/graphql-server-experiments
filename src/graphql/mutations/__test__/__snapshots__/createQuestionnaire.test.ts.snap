// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`[Mutation] createQuestionnaire can create new questionnaire 1`] = `
{
  "body": {
    "kind": "single",
    "singleResult": {
      "data": {
        "createQuestionnaire": {
          "questionnaire": {
            "description": "inventore eum sed laboriosam qui repudiandae in eum quidem illo",
            "endAt": "2021-01-31T23:59:59.000Z",
            "id": 1,
            "questions": [
              {
                "id": 1,
                "options": [
                  {
                    "id": 1,
                    "text": "Option1-1",
                  },
                  {
                    "id": 2,
                    "text": "Option1-2",
                  },
                ],
                "text": "Question1",
                "type": 1,
              },
              {
                "id": 2,
                "options": [
                  {
                    "id": 1,
                    "text": "Option2-1",
                  },
                  {
                    "id": 2,
                    "text": "Option2-2",
                  },
                ],
                "text": "Question2",
                "type": 2,
              },
            ],
            "startAt": "2021-01-01T00:00:00.000Z",
            "state": 1,
            "title": "Test questionnaire",
          },
        },
      },
      "errors": undefined,
    },
  },
  "http": {
    "headers": 
Map {
      "cache-control" => "no-store",
    }
,
    "status": undefined,
  },
}
`;
